installTerraInVm.sh

```diff
@@ -0,0 +1,53 @@
+# first make sure all terraform dependencies are installed
+# Check if gnupg is installed
+if which gpg > /dev/null 2>&1; then
+  echo 'gnupg and software-properties-common are already installed'
+else
+  echo 'Installing gnupg and software-properties-common...'
+  sudo apt-get update && sudo apt-get install -y gnupg software-properties-common
+fi
+
+# Check if the HashiCorp GPG key is present
+if gpg --no-default-keyring --keyring /usr/share/keyrings/hashicorp-archive-keyring.gpg --fingerprint | grep -q 'HashiCorp'; then
+  echo 'HashiCorp GPG key is already present.'
+else
+  echo 'HashiCorp GPG key not found. Adding the key now.'
+  wget -O- https://apt.releases.hashicorp.com/gpg | \
+  gpg --dearmor | \
+  sudo tee /usr/share/keyrings/hashicorp-archive-keyring.gpg > /dev/null
+fi
+
+# Verify the key's fingerprint
+if gpg --no-default-keyring --keyring /usr/share/keyrings/hashicorp-archive-keyring.gpg --fingerprint | grep -q 'HashiCorp'; then
```

Comment:
Maybe doing this check is a good way to check if the key is present? You can do 2 things at once by moving this check to be the check on line 11.